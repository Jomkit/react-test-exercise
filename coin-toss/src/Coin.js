import React from 'react'
import heads from "./assets/heads.jpg";
import tails from "./assets/tails.jpg";

const Coin = ({face}) => {
  return (
    <div>
        { face == "heads" ? <img src={heads} style={{width: 200}} alt={face} /> : <img src={tails} style={{width: 200}} alt={face} />}
    </div>
  )
}

export default Coin