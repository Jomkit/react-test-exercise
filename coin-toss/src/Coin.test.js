import { render } from "@testing-library/react"
import Coin from "./Coin"
// import heads from "./assets/heads.jpg"
// import tails from "./assets/tails.jpg"

// smoke test
it('should render without crashing', () => {
    render(<Coin face="heads" />);
})

// snapshot test
it("matches snapshot heads", () => {
    const { asFragment } = render(<Coin face="heads" />);
    expect(asFragment()).toMatchSnapshot();
})
it("matches snapshot tails", () => {
    const { asFragment } = render(<Coin face="tails" />);
    expect(asFragment()).toMatchSnapshot();
})