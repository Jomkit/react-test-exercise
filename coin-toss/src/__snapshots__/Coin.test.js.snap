// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot heads 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="heads"
      src="heads.jpg"
      style="width: 200px;"
    />
  </div>
</DocumentFragment>
`;

exports[`matches snapshot tails 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="tails"
      src="tails.jpg"
      style="width: 200px;"
    />
  </div>
</DocumentFragment>
`;
